analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -240,6 +256,17 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
         //       system properties "org.apache.maven.global-settings" and "org.apache.maven.user-settings".
         val settings = settingsBuilder.buildSettings()
 
+        Os.env["https_proxy"]?.let { httpsProxyUrl ->
+            // Note that even HTTPS proxies use "http://" as the protocol!
+            settings.proxies.add(createProxyFromUrl("http", httpsProxyUrl))
+            log.debug { "Added $httpsProxyUrl as proxy." }
+        }
+
+        Os.env["http_proxy"]?.let { httpProxyUrl ->
+            settings.proxies.add(createProxyFromUrl("http", httpProxyUrl))
+            log.debug { "Added $httpProxyUrl as proxy." }
+        }
```

Comment:
What happens if both `https_proxy` and `http_proxy` are set? The approach in `OkHttpClient.Builder.applyProxySettingsFromEnv()` seems better to not check `http_proxy` if `https_proxy` is set. Also, as the code in this function is very similar to the code here, I think it would be worth it to move the code that gets the proxy URL from the env variables to a separate function and reuse it here.

Reply:
I noticed this was party done in the next commit already, but I believe also the part that reads the environment variables should be generalized.

Reply:
> What happens if both `https_proxy` and `http_proxy` are set?

Whatever Maven is doing when multiple proxies are set :wink: I haven't checked TBH.

> The approach in `OkHttpClient.Builder.applyProxySettingsFromEnv()` seems better to not check `http_proxy` if `https_proxy` is set.

Well, that's simply because `OkHttp` only supports a single proxy, whereas Maven supports multiple ones.

Reply:
> I believe also the part that reads the environment variables should be generalized.

Any suggestion how? Because depending on the transport backend (OkHttp or Maven) we need to do different things with the strings from these environment variables.

Reply:
I've now made some changes, please review again.

Reply:
The way you implemented it now is what I thought of, mainly only access the env variables in a single place.

Regarding multiple proxies in Maven, the docs are not very clear. There is an `active` property:
```
active: true if this proxy is active. This is useful for declaring a set of proxies, but only one may be active at a time.
```
But this does not make it clear if there can never be multiple active proxies at the same time, or only if this property is used. Did you check if you need to set this to active in the code where you create the proxy? Or maybe `true` is the default value anyway.